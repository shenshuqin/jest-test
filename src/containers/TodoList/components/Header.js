import React ,{Component} from 'react';
import '../style.less'


class Header extends Component {
    constructor(props) {
        super(props);
        this.handleInputChange = this.handleInputChange.bind(this)
        this.handleInputKeyup = this.handleInputKeyup.bind(this)
        this.state = {
            value: ''
        }
    }
    handleInputChange (e) {
        this.setState({
            value : e.target.value
        })
    }
    handleInputKeyup(e) {
        const { value } = this.state
        if(e.keyCode === 13 && value) {
            this.props.addUndoItem(value)
        }
    }
    render() {
        const {value} = this.state;
        return (
            <div className="header">
                <div className="header-content">
                   <span> TodoList </span>
                    <input 
                        id="input" 
                        value = {value}  
                        className="header-input" 
                        placeholder="todo" 
                        onChange = {this.handleInputChange} 
                        onKeyUp = {this.handleInputKeyup}/>
                </div>
            </div> 
        )
    }
}

export default Header;