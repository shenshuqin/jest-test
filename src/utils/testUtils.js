export const findTestWrapper = (wrapper,tag) => {
    return wrapper.find(`#${tag}`)
}