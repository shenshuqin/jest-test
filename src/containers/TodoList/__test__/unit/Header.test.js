import Header from '../../components/Header'
import React from 'react'
import {findTestWrapper} from '../../../../utils/testUtils' 

import {shallow } from 'enzyme'
//第一个测试用例：浅渲染
describe('测试Header组件', function () {
    // 使用快照
    it('测试Header组件渲染样式正常', function () {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('测试Header组件包含input', function () {
        const wrapper = shallow(<Header/>);
        // const inputElem = wrapper.find("input");
        const inputElem = findTestWrapper(wrapper,'input')
        expect(inputElem.length).toBe(1);
    });
    it('测试Header组件内容初始化为空', function () {
        const wrapper = shallow(<Header/>);
        const inputElem = findTestWrapper(wrapper,'input')
        expect(inputElem.prop('value')).toEqual("");
    });
    it('测试Header组件onChange事件', function () {
        const wrapper = shallow(<Header/>);
        const inputElem = findTestWrapper(wrapper,'input')
        const userInput = '学习'
        inputElem.simulate('change',{
            target : { value : userInput}
        })
        expect(wrapper.state('value')).toEqual(userInput);
    });
    it('测试Header组件回车 ，input 无内容无操作', function () {
        const fn = jest.fn()
        const wrapper = shallow(<Header addUndoItem = {fn}/>); //当Header回车的时候触发addUndoItem
        const inputElem = findTestWrapper(wrapper,'input')
        wrapper.setState({
            value : ''
        })
        inputElem.simulate('keyUp',{
            keyCode:13
        })
        expect(fn).not.toHaveBeenCalled();
    });
    it('测试Header组件回车 ，input 有内容有操作', function () {
        const fn = jest.fn()
        const wrapper = shallow(<Header addUndoItem = {fn}/>); //当Header回车的时候触发addUndoItem
        const inputElem = findTestWrapper(wrapper,'input')
        wrapper.setState({
            value : '学习'
        })
        inputElem.simulate('keyUp',{
            keyCode:13
        })
        expect(fn).toHaveBeenCalled();
        expect(inputElem.prop('value')).toBe('')
    });
}); 