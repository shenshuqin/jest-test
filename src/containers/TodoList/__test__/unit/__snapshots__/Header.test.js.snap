// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试Header组件 测试Header组件渲染样式正常 1`] = `ShallowWrapper {}`;
